packages/netlify-cms-backend-bitbucket/src/API.ts

```diff
@@ -0,0 +1,675 @@
+  ) {
+    const formData = new FormData();
+    files.forEach(file => {
+      if ((file as DeleteEntry).delete) {
+        // delete the file
+        formData.append('files', file.path);
+      } else {
+        // add/modify the file
+        const contentBlob = get(file, 'fileObj', new Blob([(file as Entry).raw]));
+        // Third param is filename header, in case path is `message`, `branch`, etc.
+        formData.append(file.path, contentBlob, basename(file.path));
+      }
+    });
+    if (commitMessage) {
+      formData.append('message', commitMessage);
+    }
+    if (this.commitAuthor) {
+      const { name, email } = this.commitAuthor;
+      formData.append('author', `${name} <${email}>`);
+    }
+
+    formData.append('branch', branch);
+
+    if (parentSha) {
+      formData.append('parents', parentSha);
+    }
+
+    await this.request({
+      url: `${this.repoURL}/src`,
+      method: 'POST',
+      body: formData,
+    });
+
+    return files;
+  }
+
+  async persistFiles(entry: Entry | null, mediaFiles: AssetProxy[], options: PersistOptions) {
+    const files = entry ? [entry, ...mediaFiles] : mediaFiles;
+    if (options.useWorkflow) {
+      return this.editorialWorkflowGit(files, entry as Entry, options);
+    } else {
+      return this.uploadFiles(files, { commitMessage: options.commitMessage, branch: this.branch });
+    }
+  }
+
+  async addPullRequestComment(pullRequest: BitBucketPullRequest, comment: string) {
+    await this.requestJSON({
+      method: 'POST',
+      url: `${this.repoURL}/pullrequests/${pullRequest.id}/comments`,
+      headers: { 'Content-Type': APPLICATION_JSON },
+      body: JSON.stringify({
+        content: {
+          raw: comment,
+        },
+      }),
+    });
+  }
+
+  async getPullRequestLabel(id: number) {
+    const comments: BitBucketPullComments = await this.requestJSON({
+      url: `${this.repoURL}/pullrequests/${id}/comments`,
+      params: {
+        pagelen: 100,
+      },
+    });
+    return comments.values.map(c => c.content.raw)[comments.values.length - 1];
+  }
+
+  async createPullRequest(branch: string, commitMessage: string, status: string) {
+    const pullRequest: BitBucketPullRequest = await this.requestJSON({
+      method: 'POST',
+      url: `${this.repoURL}/pullrequests`,
+      headers: { 'Content-Type': APPLICATION_JSON },
+      body: JSON.stringify({
+        title: commitMessage,
+        source: {
+          branch: {
+            name: branch,
+          },
+        },
+        destination: {
+          branch: {
+            name: this.branch,
+          },
+        },
+        description: DEFAULT_PR_BODY,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        close_source_branch: true,
+      }),
+    });
+    // use comments for status labels
+    await this.addPullRequestComment(pullRequest, statusToLabel(status));
+  }
+
+  async getDifferences(branch: string) {
+    const rawDiff = await this.requestText({
```

Comment:
BitBucket provides the raw diff string so we need to parse it

Reply:
Instead of parsing the raw diff, won't the `/diffstat` endpoint be sufficient here?

Reply:
Ha, I didn't see API, let me check